src/main/java/coupon/config/DataSourceConfig.java

```diff
@@ -0,0 +1,60 @@
+package coupon.config;
+
+import com.zaxxer.hikari.HikariDataSource;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.jdbc.DataSourceBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.DependsOn;
+import org.springframework.context.annotation.Primary;
+import org.springframework.jdbc.datasource.LazyConnectionDataSourceProxy;
+
+import javax.sql.DataSource;
+import java.util.Map;
+
+@Configuration
+public class DataSourceConfig {
+
+    protected static final String WRITE = "write";
+    protected static final String READ = "read";
+
+    @Bean
+    @ConfigurationProperties(prefix = "coupon.datasource.writer")
+    public DataSource writeDataSource() {
+        return DataSourceBuilder.create()
+                .type(HikariDataSource.class)
+                .build();
+    }
+
+    @Bean
+    @ConfigurationProperties(prefix = "coupon.datasource.reader")
+    public DataSource readDataSource() {
+        return DataSourceBuilder.create()
+                .type(HikariDataSource.class)
+                .build();
+    }
+
+    @Bean
+    @DependsOn({"writeDataSource", "readDataSource"})
+    public DataSource routeDataSource() {
+        DataSource writeDataSource = writeDataSource();
+        DataSource readDataSource = readDataSource();
```

Comment:
`@DependsOn`과 같은 명시적 순서 지정이 좋을까요, 아니면 의존성 주입에서의 자동 순서 지정이 좋을까요? `@Qualifier`를 통해 어떤 이름의 빈을 가져올지 정할 수 있고, 그렇게 한다면 해당 빈이 등록된 뒤에 주입이 일어나게 됩니다. 개발자가 순서를 신경쓰면 인지부하가 커진다고 생각해서요!

```suggestion

    public DataSource routeDataSource(@Qualifier("writeDataSource") DataSource writeDataSource,
   // ...
```

Reply:
수정했습니다!~ 👍